src/test/scala/JointSpec.scala

```diff
@@ -20,65 +19,175 @@ class JointSpec extends FunSuite {
                     ((0.0, 1.0, 0.0), 1.0),      // y = 1
                     ((0.0, -1.0, 0.0), 0.0))     // -y = 0
 
-  test("Dip direction of 0 centered at origin") {
-    val joint = Joint((0.0, 0.0, 1.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0), dipAngle=0,
-                      dipDirection=0, phi=0, cohesion=0, shape=boundaries)
+  test("Dip direction for horizontal joint should be pi/2.0") {
+    val joint = Joint((0.0, 0.0, 1.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+                      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == math.Pi/2.0)
```

Comment:
So, if I'm understanding this right, in this test we expect the joint to be recognized as horizontal, and its dip direction should be set exactly to pi/2.0? If so then you're right, this will be okay.